base/tks/src/main/java/org/dogtagpki/server/tks/servlet/TokenServlet.java

```diff
@@ -1185,22 +1195,41 @@ private void processComputeSessionKey(HttpServletRequest req,
                             throw new Exception("can't generate key encryption key");
                         }
 
+                        if (aesKey != null) {
+                            logger.debug("TokenServlet: aes key encryption key generated for CUID=" +
+                                    trim(pp.toHexString(xCUID)) +
+                                    ", KDD=" +
+                                    trim(pp.toHexString(xKDD)));
+                        } else {
+                            logger.error("TokenServlet: aes key encryption key generation failed for CUID=" +
+                                    trim(pp.toHexString(xCUID)) +
+                                    ", KDD=" +
+                                    trim(pp.toHexString(xKDD)));
+
+                            throw new Exception("can't generate aes key encryption key");
+                        }
+
                         /*
                          * ECBencrypt actually takes the 24 byte DES2 key
                          * and discard the last 8 bytes before it encrypts.
                          * This is done so that the applet can digest it
                          */
 
                         byte[] encDesKey = protocol.ecbEncrypt(kek_key, desKey, selectedToken);
+			byte[] encAesKey = protocol.ecbEncrypt(kek_key,aesKey,selectedToken);
 
-                        /*
                         logger.debug("computeSessionKey:encrypted desKey size = "+encDesKey.length);
-                        logger.debug(encDesKey);
-                        */
+                        //logger.debug(encDesKey);
+
+			logger.debug("computeSessionKey:encrypted aesKey size = "+encAesKey.length);
+                        //logger.debug(encAesKey);
 
                         kek_wrapped_desKeyString =
                                 org.mozilla.jss.netscape.security.util.Utils.SpecialEncode(encDesKey);
 
+			kek_wrapped_aesKeyString =
+                               org.mozilla.jss.netscape.security.util.Utils.SpecialEncode(encAesKey);
+
                         // get keycheck
 
                         byte[] keycheck = protocol.computeKeyCheck(desKey, selectedToken);
```

Comment:
Don't you need "keycheck" for aes? shouldn't you call computeKeyCheck_SCP03() for aes key?  Where is it called?

Reply:
You are correct. That call is made in the routine computeServerSideKeyGenValues for scp03. This routine you are in is for scp01. I think I had some leftover aesKeyString, that I don't need. It is probably passed and not used, will take care of it.